shared/lib_battery_lifetime_nmc.h

```diff
@@ -0,0 +1,115 @@
+/**
+BSD-3-Clause
+Copyright 2019 Alliance for Sustainable Energy, LLC
+Redistribution and use in source and binary forms, with or without modification, are permitted provided
+that the following conditions are met :
+1.	Redistributions of source code must retain the above copyright notice, this list of conditions
+and the following disclaimer.
+2.	Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+and the following disclaimer in the documentation and/or other materials provided with the distribution.
+3.	Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+or promote products derived from this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED.IN NO EVENT SHALL THE COPYRIGHT HOLDER, CONTRIBUTORS, UNITED STATES GOVERNMENT OR UNITED STATES
+DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+OR CONSEQUENTIAL DAMAGES(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef SAM_SIMULATION_CORE_LIB_BATTERY_LIFETIME_NMC_H
+#define SAM_SIMULATION_CORE_LIB_BATTERY_LIFETIME_NMC_H
+
+#include <memory>
+
+#include "lib_util.h"
+#include "lib_battery_lifetime_calendar_cycle.h"
+
+struct lifetime_nmc_state {
+    double q_relative_li;                // %, SEI degradation
+    double q_relative_neg;               // %, cycle degradation
+    double dq_relative_li_old;
+    double dq_relative_neg_old;
+    double DOD_max;
+    int n_cycles_prev_day;
+
+    // for complex cycling of battery, b1 = summation of b1_dt * dt_day over a day
+    // lifetime capacity updated after 24 hours elapse.
+
+    double b1_dt;
+    double b2_dt;
+    double b3_dt;
+    double c2_dt;
+
+    friend std::ostream& operator<<(std::ostream& os, const lifetime_nmc_state& p);
+};
+
+class lifetime_nmc_t : public lifetime_t {
```

Comment:
Could you please add a reference to where this analytical model is from in this class description?